### PR TITLE
[HIG-3207] sessionless errors for JS backends

### PR DESCRIPTION
**Pull request by mayberryzane**
## Summary
- add `ProjectVerboseID` as an optional argument to `PushBackendPayload` and `MarkBackendSetup`
- make `SessionID` a pointer on error objects, it's already nullable in postgres
- require the verbose `projectID` to be passed in to `H.init` in the SDKs

## How did you test this change?
- tested backend errors with no sessions in the SDK handlers for `express` and `next`
- tested gql backwards compatibility by making sure existing backend `go` errors were handled correctly 

## Are there any deployment considerations?
- this is a major version bump for the `node` and `next` SDKs, will update the changelog before deploying
- changes to come for UI changes to add more details about sessionless errors on the error instance and the go SDK
- will update relevant SDK docs

### CODE REVIEW

File: sdk/highlight-node/src/handlers.ts [hunk 1]
```diff
@@ -18,18 +18,19 @@ function processErrorImpl(
 	req: { headers?: http.IncomingHttpHeaders },
 	error: Error,
 ): void {
+	let secureSessionId: string | undefined
+	let requestId: string | undefined
 	if (req.headers && req.headers[HIGHLIGHT_REQUEST_HEADER]) {
-		const [secureSessionId, requestId] =
+		;[secureSessionId, requestId] =
```
Comment (Vadman97) [inline sdk/highlight-node/src/handlers.ts:24]: neat

**Review by Vadman97 — APPROVED** [review]
nice!

File: backend/kafka-queue/types.go [hunk 2]
```diff
@@ -79,8 +80,9 @@ type PushMetricsArgs struct {
 }
 
 type MarkBackendSetupArgs struct {
-	SecureID  string
-	ProjectID int
+	SecureID         *string
+	ProjectID        int
+	ProjectVerboseID *string
```
[the remark this answers is not in the document]
Comment (mayberryzane) [reply backend/kafka-queue/types.go:85]: it wasn't being used as far as I could tell, will remove

File: backend/public-graph/graph/schema.resolvers.go [hunk 3]
```diff
@@ -125,20 +125,28 @@ func (r *mutationResolver) PushPayload(ctx context.Context, sessionSecureID stri
 }
 
 // PushBackendPayload is the resolver for the pushBackendPayload field.
-func (r *mutationResolver) PushBackendPayload(ctx context.Context, errors []*customModels.BackendErrorObjectInput) (interface{}, error) {
-	errorsBySecureID := map[string][]*customModels.BackendErrorObjectInput{}
+func (r *mutationResolver) PushBackendPayload(ctx context.Context, projectID *string, errors []*customModels.BackendErrorObjectInput) (interface{}, error) {
+	errorsBySecureID := map[*string][]*customModels.BackendErrorObjectInput{}
 	for _, backendError := range errors {
 		errorsBySecureID[backendError.SessionSecureID] = append(errorsBySecureID[backendError.SessionSecureID], backendError)
```
[the remark this answers is not in the document]
Comment (mayberryzane) [reply backend/public-graph/graph/schema.resolvers.go:131]: hmm I don't think so, there should be a `nil` map entry e.g.:

Screen Shot 2023-01-30 at 11 24 48 AM

also when I tested locally this was working correctly, I can test some stuff in prod

**Pull request merged**